crates/argmin/src/core/executor.rs

```diff
@@ -383,7 +386,7 @@ where
         self
     }
 
-    /// Enables or disables timing of individual iterations (default: enabled).
+    /// Enables or disables timing of individual iterations (default: when needed).
     ///
     /// Setting this to false will silently be ignored in case a timeout is set.
```

Comment:
```suggestion
    /// Enables or disables timing of individual iterations (default: false).
    ///
    /// In case a timeout is set, this will automatically be set to true.
```

Now that I think of it, I think this should probably be set to `false` by default (independent of whether observers are used or not), as it makes documentation substantially easier.